src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -89,81 +124,453 @@ static DamageTable D_808B10C8 = {
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Am/EnAm_Draw.s")
+extern AnimationHeader D_06000238; // Wake up
+extern AnimationHeader D_0600033C; // Hop Anim
+extern AnimationHeader D_06005B3C; // Take Damage
+extern SkeletonHeader D_06005948;
+
+void EnAm_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnAm* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 42.857143f);
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_06005948, &D_0600033C, this->jointTable, this->morphTable, 14);
+    Collider_InitAndSetCylinder(globalCtx, &this->enemyCollider, &this->actor, &sEnemyCylinderInit);
+    Collider_InitAndSetCylinder(globalCtx, &this->interactCollider, &this->actor, &sCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+    this->actor.home.pos.x -= 9.0f * Math_SinS(this->actor.shape.rot.y);
+    this->actor.home.pos.z -= 9.0f * Math_CosS(this->actor.shape.rot.y);
+    this->actor.world.pos.x = this->actor.home.pos.x;
+    this->actor.world.pos.z = this->actor.home.pos.z;
+    func_808AFF9C(this);
+}
+
+void EnAm_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnAm* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->enemyCollider);
+    Collider_DestroyCylinder(globalCtx, &this->interactCollider);
+}
+
+// EnAm_Hop
+void func_808AFE38(EnAm* this, GlobalContext* globalCtx) {
+    s32 i;
+    Vec3f pos;
+    s32 pad;
+
+    for (i = 4; i > 0; i--) {
+        pos.x = randPlusMinusPoint5Scaled(65.0f) + this->actor.world.pos.x;
+        pos.y = randPlusMinusPoint5Scaled(10.0f) + (this->actor.world.pos.y + 40.0f);
+        pos.z = randPlusMinusPoint5Scaled(65.0f) + this->actor.world.pos.z;
+        EffectSsKiraKira_SpawnSmall(globalCtx, &pos, &sVelocity, &sAccel, &D_808B1118, &D_808B111C);
+    }
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_AMOS_WALK);
+    func_800BBDAC(globalCtx, &this->actor, &this->actor.world.pos, 4.0f, 3, 8.0f, 0x12C, 0xF, 0);
+}
+
+void func_808AFF9C(EnAm* this) {
+    f32 lastFrame = Animation_GetLastFrame(&D_0600033C);
+
+    Animation_Change(&this->skelAnime, &D_0600033C, 0.0f, lastFrame, lastFrame, 0, 0.0f);
+    this->enemyCollider.info.bumper.dmgFlags = 0x80000088;
+    this->interactCollider.info.bumper.dmgFlags = 0x77CFFF77;
+    if (this->actor.colChkInfo.health != 0) {
+        this->enemyCollider.base.atFlags &= ~1;
+    }
+    Math_Vec3f_Copy(&this->actor.world.pos, &this->actor.home.pos);
+    this->actionFunc = func_808B0040;
+}
+
+void func_808B0040(EnAm* this, GlobalContext* globalCtx) {
+    if (((this->enemyCollider.base.ocFlags1 & OC1_HIT) && (this->enemyCollider.base.ocFlags2 & OC2_HIT_PLAYER)) ||
+        (this->interactCollider.base.acFlags & AC_HIT)) {
+        if (this->unk_234 == 0) {
+            func_808B00D8(this);
+        }
+        return;
+    }
+    if (this->unk_234 >= 0xB) {
+        this->unk_234 -= 0xA;
```

Comment:
```suggestion
    if (this->unk_234 > 10) {
        this->unk_234 -= 10;
```
I think this was commented in the other review, but all instances of `unk_234` can be in dec